Iris compatibility

marc0olo opened the pull request.

mradkov commented:
Can you update the sdk to 8.0.0?

mradkov commented on package.json:
```diff
@@ -11,7 +11,7 @@
   },
   "dependencies": {
-    "@aeternity/aepp-sdk": "7.7.0",
+    "@aeternity/aepp-sdk": "8.0.0-beta.2",
```
```suggestion
    "@aeternity/aepp-sdk": "^8.0.0",
```

marc0olo replied:
yeah I will soon commit a new state including the updated SDK

marc0olo commented:
@mradkov if desired we could publish a first beta version based on that state before we target additional changes and the bigger refactoring

marc0olo commented:
@mradkov short additions:
- hostedCompiler property is currently pointing to `https://v500.compiler.aeternity.art` -> should be changed
- the "deploy" functionality is currently not implemented -> I dropped the usage of the aeproject specific deployer

marc0olo commented:
@mradkov I made some further cleanups and updated the node version to v6.0.0-rc3. now the init command also includes a working example deployment script again.

I also removed some commands and the respective logic:
- shapes
- contracts
- aestudio

I am not sure whether we need these commands for local development with aeproject. if so we need to make sure everything works. so I thought it's better to remove it at this point of time.

mradkov commented:
> I am not sure whether we need these commands for local development with aeproject. if so we need to make sure everything works. so I thought it's better to remove it at this point of time.

I'm also unaware if anyone is even using those. I like the approach of dropping them for now and reintroducing them later with improved architecture and reliability in case they are needed.

mradkov commented:
> @mradkov short additions:
> 
> * hostedCompiler property is currently pointing to `https://v500.compiler.aeternity.art` -> should be changed
> * the "deploy" functionality is currently not implemented -> I dropped the usage of the aeproject specific deployer

Are these two addressed now?

marc0olo commented:
> > @mradkov short additions:
> > 
> > * hostedCompiler property is currently pointing to `https://v500.compiler.aeternity.art` -> should be changed
> > * the "deploy" functionality is currently not implemented -> I dropped the usage of the aeproject specific deployer
> 
> Are these two addressed now?

yes I changed that to the https://latest.compiler.aepps.com and I provided a sample for deployment which should work.

regarding the compiler I should probably update to `5.1.0` now as we have the new release including contract factories:
- https://github.com/aeternity/aesophia_http/releases/tag/v5.1.0

mradkov commented:
> > > @mradkov short additions:
> > > 
> > > * hostedCompiler property is currently pointing to `https://v500.compiler.aeternity.art` -> should be changed
> > > * the "deploy" functionality is currently not implemented -> I dropped the usage of the aeproject specific deployer
> > 
> > 
> > Are these two addressed now?
> 
> yes I changed that to the https://latest.compiler.aepps.com and I provided a sample for deployment which should work.
> 
> regarding the compiler I should probably update to `5.1.0` now as we have the new release including contract factories:
> 
> * https://github.com/aeternity/aesophia_http/releases/tag/v5.1.0

Is `5.1.0` already hosted?

marc0olo commented:
> > > > @mradkov short additions:
> > > > 
> > > > * hostedCompiler property is currently pointing to `https://v500.compiler.aeternity.art` -> should be changed
> > > > * the "deploy" functionality is currently not implemented -> I dropped the usage of the aeproject specific deployer
> > > 
> > > 
> > > Are these two addressed now?
> > 
> > 
> > yes I changed that to the https://latest.compiler.aepps.com and I provided a sample for deployment which should work.
> > regarding the compiler I should probably update to `5.1.0` now as we have the new release including contract factories:
> > 
> > * https://github.com/aeternity/aesophia_http/releases/tag/v5.1.0
> 
> Is `5.1.0` already hosted?

not yet I guess. just checked by providing the version header to https://compiler.aepps.com/version. but for local development I already activated it now

thepiwo commented on aeproject-cli/aeproject-init/artifacts/docker/aeternity.yaml:
```diff
@@ -21,15 +18,6 @@ keys:
-    "1": 0
-    "2": 1
-    "3": 2
-    "4": 3
```
shouldn't this still be here to use the latest protocol?

also persistence for when using the same container again

marc0olo replied:
will check that, don't think so. persistence will be added

thepiwo commented on aeproject-cli/aeproject-init/artifacts/exampleTests.js:
```diff
@@ -0,0 +1,72 @@
+ *  PERFORMANCE OF THIS SOFTWARE.
+ */
+const fs = require('fs')
+const { Universal: Ae, MemoryAccount, Node } = require('@aeternity/aepp-sdk');
```
why alias universal?

marc0olo replied:
that was probably a copy paste from elsewhere. noticed this yesterday in the faucet-migration, too. will remove the alias.

thepiwo commented on aeproject-cli/aeproject-init/artifacts/exampleTests.js:
```diff
@@ -0,0 +1,72 @@
+
+    before(async () => {
+        const account = MemoryAccount({ keypair: wallets[0] });
+        const node = await Node({ url: NETWORKS[NETWORK_NAME].nodeUrl, internalUrl: NETWORKS[NETWORK_NAME].nodeUrl });
```
internalUrl needs no specification if same as url

thepiwo commented on aeproject-cli/aeproject-init/artifacts/exampleTests.js:
```diff
@@ -0,0 +1,72 @@
+    let hamsterName;
+
+    before(async () => {
+        const account = MemoryAccount({ keypair: wallets[0] });
```
where is wallets coming from here, I propose to make this more of an explicit import

marc0olo replied:
I would prefer to leave it as is right now and address this topic in the bigger refactoring. this is related to the funded wallets and they are provided in the tests through this line: https://github.com/marc0olo/aepp-aeproject-js/blob/6ede4d647b80131307f8b2096923677a44705b2d/aeproject-cli/aeproject-test/aeproject-test.js#L76

marc0olo replied:
what way would you prefer here? we need to consider:
- funding the wallets
- providing possibility to access the funded wallets in tests

ideally the dev could provide a mnemonic to use. but I want to avoid having the dev to define that twice for different cli commands. would this be a way to go in order to have that implemented quickly?
- use the mnemonic based on an ENV variable -> fallback to default mnemonic in case it's missing
    - would be used for funding and in the tests
- provide a new command that allows to fund additionall wallets quickly based on a mnemonic and additionally providing the amount of derivations

thepiwo replied:
I was just thinking about explicit import instead of magic aeproject stuff.

as done here: https://github.com/thepiwo/aeproject-boilerplate/blob/master/test/exampleTest.js#L3

thepiwo replied:
same for the chai assert probably.

then IDEs can use their native mocha integration for testing

marc0olo replied:
ok I mean if this "overwrites" the magic aeproject stuff I can add that quickly. everything else should be addressed in the bigger refactoring.

as we don't release a separate package anymore I will add that json file to the new config folder where I also placed the networks and refer to this in the exampleTest.

would this be ok?

marc0olo replied:
I guess I need to populate the package.json of the default template with chai and mocha, too. right?

marc0olo replied:
see recent commit. are we fine with that? :)

thepiwo replied:
looks good as far as I can see, haven't tried it out

thepiwo commented on aeproject-cli/aeproject-init/artifacts/exampleTests.js:
```diff
@@ -0,0 +1,72 @@
+              { name: NETWORK_NAME, instance: node },
+            ],
+            compilerUrl: NETWORKS[NETWORK_NAME].compilerUrl,
+            accounts: [account],
```
I prefer inline MemoryAccount definition, makes it more copy-pasteable

thepiwo commented on aeproject-cli/aeproject-init/artifacts/exampleTests.js:
```diff
@@ -0,0 +1,72 @@
+    it('Should return false if name does not exist', async () => {
+        hamsterName = 'DoesHamsterExists';
+        let result = await contract.methods.nameExists(hamsterName);
+        assert.isOk(!result.decodedResult);
```
`isOk(!` ?

marc0olo replied:
not my code xD ... I will change it^^

thepiwo replied:
I know :D just keep the examples clean, others will copy

marc0olo replied:
definitely agree. every optimization is good :) - though we are aiming for a bigger refactoring anyway

thepiwo commented on aeproject-cli/aeproject-init/artifacts/exampleTests.js:
```diff
@@ -0,0 +1,77 @@
+var chai_as_promised = require("chai-as-promised");
+chai.use(chai_as_promised);
+const assert = chai.assert;
+const expect = chai.expect;
```
lets just use one chai syntax, I prefer assert over expect

marc0olo replied:
see comment above

thepiwo commented on aeproject-cli/aeproject-init/artifacts/exampleTests.js:
```diff
@@ -0,0 +1,77 @@
+const { Universal, MemoryAccount, Node } = require('@aeternity/aepp-sdk');
+
+const EXAMPLE_CONTRACT = fs.readFileSync('./contracts/ExampleContract.aes', 'utf8');
+const NETWORKS = require('../config/network.json');
```
why move that into every project instead of aeproject dependency?

marc0olo replied:
mhh, as devs install aeproject globally to use the CLI I wanted to avoid having to add aeproject in each project, too. this way it can also be modified easily. not sure what's better here

marc0olo replied:
as discussed I will leave that as is right now. certain files and functions will be copied into every new project for the moment. I also added contract-utils that can determine the `filesystem` object which needs to be provided to the compiler for custom includes. the provided example now also contains a custom include.

thepiwo commented on aeproject-cli/aeproject-init/artifacts/exampleTests.js:
```diff
@@ -0,0 +1,77 @@
+ */
+const fs = require('fs');
+const chai = require('chai');
+var chai_as_promised = require("chai-as-promised");
```
lets just use await for promises instead of some additional dependency

marc0olo replied:
alright. I added it to keep the tests 1:1. it seems like assert.isFulfilled doesn't exist anymore in the current version of chai so I added expect here. do you know the alternative for assert? otherwise I would just remove the line that checks if the promise resolves

thepiwo commented:
- keys in docker setup can be removed, also key config in aeternity.yaml
 - cors nginx config is empty and not needed
 - ws and other nginx configs can be merged in one file (if even needed)
 - entrypoint and healthcheck scripts are unused and can be removed
 - lots of unused code can be removed, e.g. readFileRelative from old utilities
 - compiler and node can be moved to same docker compose file

marc0olo commented:
yeah I will address as much of these topics as possible shortly. but I wouldn't want to waste too much time in this stuff as we will probably perform a bigger refactoring anyway

marc0olo commented:
> * keys in docker setup can be removed, also key config in aeternity.yaml
> * cors nginx config is empty and not needed
> * ws and other nginx configs can be merged in one file (if even needed)
> * entrypoint and healthcheck scripts are unused and can be removed
> * lots of unused code can be removed, e.g. readFileRelative from old utilities
> * compiler and node can be moved to same docker compose file

**Done**
- removed unused files
    - keys
    - cors & ws nginx config
    - entrypoint & healthcheck scripts

**Open**
- remove unused code (there is too much cleanup required, don't want to waste more time right now)
- merge docker-compose file of compiler and node
   - there is some functionality which provides prompts to the user in case of updating a project. I haven't checked & tested it so not sure if it works
   - sometimes it makes sense to run the compiler and the node independently. I know this can be done using one file but that would require additional modifications to aeprojects internals which I don't want to waste more time on right now

marc0olo commented:
@mradkov @thepiwo I think now we can publish a release. I assumed that we cannot use the gitbook of the old maintainers and added the ToC of the documentation directly to the main README. additionally I added some guidance how to migrate from old projects created with AEproject
